### DISCUSSION TITLE
config dap.configurations.java

### DISCUSSION THREAD

**Discussion by vozfrezz** [magidc/nvim-config#2] [Q&A]
Hi pro.  I'm a beginner with Neovim. Please help me configure debugging for Java programs. I'm a Windows OS user.

**Comment by vozfrezz** [comment]
oh men. i have done this.

**Reply by magidc** [reply]
Hi, how can I help you?
All my configuration for Java LSP and debugging is here [jdtls.lua](https://github.com/magidc/nvim-config/blob/master/lua/lsp/configs/jdtls.lua). In the init_options field of jdtls config structure I am attaching a bundle with the path to the JAR library of VSCode Java Debug. In my case, as I downloaded the file using a manager plugin (mason) I am using it to get the path, make sure you have executed :MasonInstall java-debug-adapter and :MasonInstall java-test
